### PR TITLE
support long path names for the app-data seeder generates console entry points

### LINKED ISSUE #997: Long env path names cause weird errors after activiting virtual environment
After creating a virtualenv in the path

`/home/harald/git-repos/sandbox/stackoverflow/41454203-how-do-you-import-the-library-being-tested-when-running-a-nosetest/env`

I could not run the env's pip,

```
nosetest/env$ ./bin/pip
bash: ./bin/pip: /home/harald/git-repos/sandbox/stackoverflow/41454203-how-do-you-import-the-l: bad interpreter: Permission denied
```

and running just pip gives me the system's pip, with all of its packages.

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves #997.

### CODE REVIEW

File: tests/unit/create/test_creator.py [hunk 1]
```diff
@@ -297,3 +297,18 @@ def test_creator_input_passed_is_abs(tmp_path, monkeypatch):
     monkeypatch.chdir(tmp_path)
     result = Creator.validate_dest("venv")
     assert str(result) == str(tmp_path / "venv")
+
+
+def test_create_long_path(current_fastest, tmp_path):
+    if sys.platform == "darwin":
+        max_shebang_length = 512
+    else:
+        max_shebang_length = 127
```
Comment (systemlogic) [inline tests/unit/create/test_creator.py]: this should also be 512. Our builds are genarating longer path one for example
```
#!/home/gcdo/.cache/bazel/_bazel_gcdo/7973012a3ebb4144bd899078c66d5296/execroot/com_ibm_monorepo/bazel-out/k8-fastbuild/bin/cedp/web/rest_bzl_pipenv_interpreter.sh.runfiles/.venv/bin/python
# -*- coding: utf-8 -*-
import re
import sys

from pip._internal.cli.main import main

if __name__ == "__main__":
    sys.argv[0] = re.sub(r"(-script.pyw?|.exe)?$", "", sys.argv[0])
    sys.exit(main())
```

Comment (gaborbernat) [reply tests/unit/create/test_creator.py]: These are the limits as taken from ``distlib`` of how it triggers the long-path special casing; those there are only used for tests.

Comment (asottile) [reply tests/unit/create/test_creator.py]: the test is correct, on posixlike platforms the shebang limit is 127 characters

**Comment by pradyunsg** [comment]
Any reason to not switch make_exe to use ScriptMaker?

**Comment by gaborbernat** [comment]
it does now 👍

**Pull request merged**